src/Symfony/Component/Lock/LockInterface.php

```diff
@@ -23,12 +23,13 @@
 interface LockInterface
 {
     /**
-     * Acquires the lock. If the lock is acquired by someone else, the parameter `blocking` determines whether or not
+     * Acquires the lock. If the lock is acquired by someone else, the parameter `blocking` determines whether
      * the call should block until the release of the lock.
      *
      * @return bool
      *
-     * @throws LockConflictedException If the lock is acquired by someone else in blocking mode
+     * @throws LockConflictedException If the lock is acquired by someone else or if the underlying blocking lock
+     *                                 store couldn't acquire the lock in blocking mode
```

Comment:
Not being able to acquire the lock is precisely what happens when it is already acquired by someone else.

Reply:
and if we cannot acquire it due to a technical issue (Redis not being able to work), we should be getting a LockAcquiringException described below, not a LockConflictedException. So I'm -1 for this change.

Reply:
It seems that the logic in some blocking stores doesn't reflect that. To take an example, here's what the author told in its issue:

> In the case of Flock, that's when PHP's flock() returns false, which is not when the lock is already acquired by someone else, because in that case it would block and wait. I'm not sure what an example would be of such a case.

Same with `RetryTillSaveStore`, this exception is thrown if the lock couldn't be acquired after a given number of tries (which could be the result of "anything").

Reply:
But maybe the real fix is to change the thrown exception in `RetryTillSaveStore` for example, which would make more sense. But I'm afraid it would be a breaking change?

Reply:
Well, when RetryTillSaveStore reaches the maximum number of retries for its blocking behavior, LockConflictedException makes sense as it means that the lock is still held by someone else when the end of the timeout is reached.

side note, RetryTillSaveStore has been removed.

Reply:
Okay I got your point, closing then